website/blog/analytics-craft/2022-02-07-customer-360-view-census-playbook.md

```diff
@@ -0,0 +1,404 @@
+
+
+```
+{% macro extract_email_domain(email) %}
+
+{# This is the SQL to extract the email domain in the Snowflake Flavor of SQL #}
+
+	regexp_substr(lower({{ email }}), '@(.*)', 1, 1, 'e',1)
+
+{% endmacro %}
+```
+
+> Builder Beware! Notice we didn’t check for improperly formatted emails, like periods at the end of the domain or whitespaces. Make sure you check your dataset to see if this is a valid assumption.
+
+Generally, it’d be useful to leverage a regular expression to strip and pull down an email address. However, because this is a B2B use case, not all email domains are created equal. We want to make sure we flag personal emails so they’re treated differently than the corporate emails our sales team will reach out to (this makes sales outreach more productive, and ensures we aren’t contacting people more than once).
+
+**Tip:** If you’re building out a definition like "personal email domains" for the first time, I strongly recommend building alignment upfront with the rest of the business. . Understanding the impact and having a shared understanding of these kinds of definitions reduces friction and allows you to [run your data team like a product team](https://locallyoptimistic.com/post/run-your-data-team-like-a-product-team/) rather than responding to ad hoc service requests.
+
+### Step 2.2: Flag personal emails
+
+Next, we can **flag personal emails** with [`models/jafflegaggle_contacts.sql`](https://github.com/dflynn20/jaffle_gaggle/blob/main/models/jafflegaggle_contacts.sql), which calls another macro at the top of the file to pull in the personal emails we would like to exclude:
+
+```
+{% macro get_personal_emails() %}
+
+	{{ return(('gmail.com', 'outlook.com', 'yahoo.com', 'icloud.com', 'hotmail.com')) }}
+
+{% endmacro %}
+```
+
+One of the great things about writing this as a macro in dbt is that the data team can easily reuse this file in other places in the codebase.
+
+Doing so improves consistency and makes sure additions or deletions to this personal email list are up to date.
+
+### Step 2.3: Create a column for corporate email
+
+Next, we’ll **create a column for corporate email** that will be null if an email domain is personal, also in the same `jafflegaggle_contacts` model:
+
+```
+iff(users.email_domain in {{ personal_emails }}, null, users.email_domain)
+
+           as corporate_email
+```
+
+> Builder beware! Not all of these treatments are exhaustive. You might encounter country suffixes on email domains or other domains entirely. Be sure to check for your own use case, and add the columns that make the most sense in your scenario.
+
+The other aspects to this users model are related to the event data we reference in the event stream. For example, the `order_placed` event is broken out in a CTE because it’s important to our use case at JaffleGaggle (it’s the basis for getting that dough 💰).
```

Comment:
My brain keeps sliding off this sentence but I'm having trouble figuring out how to rephrase it